priv/rest.erl.eex

```diff
@@ -154,14 +162,17 @@ handle_response({ok, StatusCode, ResponseHeaders, Client}, SuccessStatusCode)
                         StatusCode =:= SuccessStatusCode ->
             {ok, #{}, {StatusCode, ResponseHeaders, Client}};
         {ok, Body} ->
-            Result = <%= context.decode %>,
+            Result = case DecodeBody of
+                       true -> <%= context.decode %>;
+                       false -> #{<<"Body">> => Body}
```

Comment:
Apparently, the code expects this to be a map. The generated Erlang code for an API looks like this:

```
    case request(Client, Method, Path, Query_, Headers, Input, Options, SuccessStatusCode) of
      {ok, Body0, {_, ResponseHeaders, _} = Response} ->
        ResponseHeadersParams =
          [
            {<<"x-amz-server-side-encryption-bucket-key-enabled">>, <<"BucketKeyEnabled">>},
            [...]
            {<<"x-amz-version-id">>, <<"VersionId">>}
          ],
        FoldFun = fun({Name_, Key_}, Acc_) ->
                      case lists:keyfind(Name_, 1, ResponseHeaders) of
                        false -> Acc_;
                        {_, Value_} -> Acc_#{Key_ => Value_}
                      end
                  end,
        Body = lists:foldl(FoldFun, Body0, ResponseHeadersParams),
        {ok, Body, Response};
      Result ->
        Result
    end.
```

So, it looks like we merge the "body" with the "response headers". This does not look too nice, also not sure `<<"Body">>` is the key we want. But I guess this should be addressed separately from this PR. Any insights @jfacorro ?